src/main/java/nl/tudelft/lifetiles/graph/models/sequence/SequenceSegment.java

```diff
@@ -154,6 +160,110 @@ public final long distanceTo(final SequenceSegment other) {
      */
     @Override
     public final int compareTo(final SequenceSegment other) {
-        return Long.compare(this.getStart(), other.getStart());
+        int candidateComp = Long.compare(this.getAbsStart(),
```

Comment:
After googling around a bit, I've learned that big `if`-chains like this result in bad testability. The general way to solve this is to use polymorphism (or the command pattern):

- Create an interface with 1 method
- For each `if`-clause create an implementation (either as inner class, anonymous class or lambda) and put their instances in a list. 
- Loop over the list and each iteration and break if the candidate `!= 0`. 

Now you can simply test by executing the implementations individually instead of having to craft the sequence segments so that all if branches are reached.

Google tech talk explaining polymorphism & testing: https://youtu.be/4F72VULWFvc